rust/chains/hyperlane-cardano/src/multisig_ism.rs

```diff
@@ -0,0 +1,67 @@
+use std::str::FromStr;
+
+use crate::provider::CardanoProvider;
+use crate::rpc::CardanoRpc;
+use crate::ConnectionConf;
+use async_trait::async_trait;
+
+use hyperlane_core::{
+    ChainResult, ContractLocator, HyperlaneChain, HyperlaneContract, HyperlaneDomain,
+    HyperlaneMessage, HyperlaneProvider, MultisigIsm, H256,
+};
+
+/// MultisigIsm contract on Cardano
+#[derive(Debug)]
+pub struct CardanoMultisigIsm {
+    cardano_rpc: CardanoRpc,
+    domain: HyperlaneDomain,
+}
+
+impl CardanoMultisigIsm {
+    /// Create a new Cardano CardanoMultisigIsm
+    pub fn new(conf: &ConnectionConf, locator: ContractLocator) -> Self {
+        let cardano_rpc = CardanoRpc::new(&conf.url);
+        Self {
+            cardano_rpc,
+            domain: locator.domain.clone(),
+        }
+    }
+}
+
+impl HyperlaneChain for CardanoMultisigIsm {
+    fn domain(&self) -> &HyperlaneDomain {
+        &self.domain
+    }
+
+    fn provider(&self) -> Box<dyn HyperlaneProvider> {
+        Box::new(CardanoProvider::new(self.domain.clone()))
+    }
+}
+
+impl HyperlaneContract for CardanoMultisigIsm {
+    fn address(&self) -> H256 {
+        // ISM on Cardano is a minting policy, not an address
+        // We could return the minting policy hash here?
+        !todo!()
+    }
+}
+
+#[async_trait]
+impl MultisigIsm for CardanoMultisigIsm {
+    /// Returns the validator and threshold needed to verify message
+    async fn validators_and_threshold(
+        &self,
+        message: &HyperlaneMessage,
+    ) -> ChainResult<(Vec<H256>, u8)> {
+        // We're using the same multisig ISM for all messages
+        // TODO: https://github.com/tvl-labs/hyperlane-cardano/issues/42
+        // will enable dApp-defined ISM
+        let parameters = self.cardano_rpc.get_ism_parameters().await.unwrap();
```

Comment:
Can we handle this error and not make it panic?

Reply:
We can do it after the MVP for the whole app, currently it's the norm everywhere:
- https://github.com/tvl-labs/hyperlane-monorepo/pull/3/files#diff-f19dbfe4182f95fe54e033dc8e1a6d970807c66b1b4f22208c268673e555c73fR62
- https://github.com/tvl-labs/hyperlane-cardano/issues/59